packages/ssi-types/src/mapper/credential-mapper.ts

```diff
@@ -293,6 +293,82 @@ export class CredentialMapper {
     return '@context' in credential && ((credential as ICredential).type?.includes('VerifiableCredential') || false)
   }
 
+  public static isCredential(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): boolean {
+    let document = original
+    try {
+      if (typeof original === 'string') {
+        if (this.isJwtEncoded(original)) {
+          const vc: IVerifiableCredential = this.toUniformCredential(original)
+          return this.isW3cCredential(vc)
+        } else if (this.isSdJwtEncoded(original)) {
+          // find out how we can differentiate between vc and vp for sd-jwts
+          return true
+        } else {
+          document = JSON.parse(original)
+        }
+      }
+      return (
+        this.isW3cCredential(document as ICredential) ||
+        this.isSdJwtDecodedCredentialPayload(document as ICredential) ||
+        this.isJwtDecodedCredential(document as OriginalVerifiableCredential) ||
+        this.isSdJwtDecodedCredential(document as OriginalVerifiableCredential)
+      )
+    } catch (e) {
+      return false
+    }
+  }
+
+  public static isPresentation(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): boolean {
+    let document = original
+    try {
+      if (typeof original === 'string')
+        if (this.isJwtEncoded(original)) {
+          const vp: IVerifiablePresentation = this.toUniformPresentation(original)
+          return this.isW3cPresentation(vp)
+        } else if (this.isSdJwtEncoded(original)) {
+          return true
+        } else {
+          document = JSON.parse(original)
+        }
+      return (
+        this.isW3cPresentation(document as IPresentation) ||
+        this.isSdJwtDecodedCredentialPayload(document as ICredential) ||
+        this.isJwtDecodedPresentation(document as OriginalVerifiablePresentation) ||
+        this.isSdJwtDecodedCredential(document as OriginalVerifiableCredential)
+      )
+    } catch (e) {
+      return false
+    }
+  }
+
+  public static hasProof(original: OriginalVerifiableCredential | OriginalVerifiablePresentation | string): boolean {
+    let document = original
+    try {
+      if (this.isJwtEncoded(original) || this.isJwtDecodedCredential(original as OriginalVerifiableCredential)) {
```

Comment:
Hmzz, this behaves different from the getFirstProof method, which is now made public. The latter returns undefined for a JWT, suggesting there is no proof (that of course is debatable, because we really should look at whether a signature is present in the jwt (simply split the jwt by '.' and look at whether the 3rd index is present.

Whether that should be added to the getFirstProof method, or whether this method should behave the same as getFirstProof is not something I can answer from this quick review, because I need to know the context from which the getFirstProof method was called in the past.

I am making a fuss about it, because we now have 2 public methods related to proofs. One called hasProof which for a JWT returns true, but then the getFirstProof returns undefined for that same JWT. Simply adjusting the latter function might have undesired effects on existing code that called the previous internal function.

Reply:
I personally think the impl is wrong on the getFirstProof method. jwt encoded credentials have been signed and we can get the signature the way you have described. but exactly because of your last point I didn't change the existing impl. I'm also confused about the `decodeVerifiableCredential` which is creating an embedded proof for the decoded credential and passing in the whole document in the jwt part of the proof.

Reply:
Without looking at it I already tend to disagree about it being wrong. Why? The method was private for a reason and thus is constrained to a very specific use-case. That use-case is probably either not involving JWTs or has some other detections going on. The word Proof can be interpreted on that method in 2 ways. One is that a signature should be present (which is how you are now using it), the other simply is litteraly about the Proof object being present in the VC or not. 

The problem happens because you made the method public, and then of course the expectations could be completely different. So you first need to look at the code path(s) calling the previously private method, to determine whether the method was wrong to begin with or not; whether it assumed a signature or proof object. Only then you can decide whether the method needs changes or not when it needs to become public.

Regarding the decoding and passing a jwt part. That is literally how we and Veramo create a uniform representation, transforming a JWT string into a json ld like credential, allowing access to the claims. Then by adding a constructed proof, with value JwtProof2020 and passing in the full JWT VC, you have a Proof method that can be verified using the original JWT and you have a VC that can easily be used just like any other JSONLD VC

Reply:
I reverted and made the `getFirstProof` private again. I'm using the functions `toUniformCredential` and `toUniformPresentation` functions in my mapper, and it's using this function and later I'm getting the proof value from the resulted `IVerifiableCredential`/`IVerifiablePresentation`